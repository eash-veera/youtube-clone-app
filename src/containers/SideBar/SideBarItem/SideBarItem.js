import React from 'react';
import { Icon , Menu } from 'semantic-ui-react';
import './SideBarItem.scss';

export function SideBarItem(props) {
    const heightlight = props.highlight ? 'highlight-item' : null;

    return (
        <Menu.Item className={['sidebar-item', heightlight].join(' ')}>
            <div className='sidebar-item-alignment-container'>
                <span><Icon size='large' name={props.icon} /></span>
                <span>{props.label}</span>
            </div>
        </Menu.Item>
    );
}