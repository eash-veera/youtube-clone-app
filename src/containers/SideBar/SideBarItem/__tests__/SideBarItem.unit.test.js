import React from 'react';
import { shallow } from 'enzyme';
import { SideBarItem } from '../SideBarItem';

describe('SideBarItem', () => {
    //test for rendering empty SideBarItem
    test('Renders empty SideBarItem', () => {
        const wrapper = shallow(
            <SideBarItem />
        );
        expect(wrapper).toMatchSnapshot();
    });

    //test for rendering highlighted SideBarItem which navigates to /feed/trending
    test('Renders highlighted SideBarItem which navigates to /feed/trending', () => {
        const wrapper = shallow(
            <SideBarItem highlighted icon='fire' label='Trending'/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    //test for rendering non-highlighted SideBarItem which navigates to /feed/trending
    test('Renders non-highlighted SideBarItem which navigates to /feed/trending', () => {
        const wrapper = shallow(
            <SideBarItem icon='fire' label='Trending'/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    //test for rendering non-highlighted SideBarItem which navigates to /feed/trending
    test('Renders highlighted SideBarItem which no navigation', () => {
        const wrapper = shallow(
            <SideBarItem highlighted icon='fire' label='Trending'/>
        );
        expect(wrapper).toMatchSnapshot();
    });
});