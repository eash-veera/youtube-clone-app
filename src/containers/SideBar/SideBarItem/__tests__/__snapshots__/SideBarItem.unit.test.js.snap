// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarItem Renders empty SideBarItem 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        size="large"
      />
    </span>
    <span />
  </div>
</MenuItem>
`;

exports[`SideBarItem Renders highlighted SideBarItem which navigates to /feed/trending 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="fire"
        size="large"
      />
    </span>
    <span>
      Trending
    </span>
  </div>
</MenuItem>
`;

exports[`SideBarItem Renders highlighted SideBarItem which no navigation 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="fire"
        size="large"
      />
    </span>
    <span>
      Trending
    </span>
  </div>
</MenuItem>
`;

exports[`SideBarItem Renders non-highlighted SideBarItem which navigates to /feed/trending 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="fire"
        size="large"
      />
    </span>
    <span>
      Trending
    </span>
  </div>
</MenuItem>
`;
